### PR TITLE
Add docs on self-hosted CSP

### PR DESCRIPTION
**Pull request by oioki**

### CODE REVIEW

File: src/docs/self-hosted/csp.mdx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+</Alert>
+
+Starting with Sentry `23.5.0`, it is possible to enable [CSP header](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) on the self-hosted Sentry installation. The good news is that Sentry itself supports [CSP reports collection](https://docs.sentry.io/product/security-policy-reporting/). We recommend creating a separate Sentry project for CSP reports. To enable CSP and reports collection, you'll want to configure the following settings in `sentry.conf.py`:
+
+```
+if "csp.middleware.CSPMiddleware" not in MIDDLEWARE:
+    MIDDLEWARE = ("csp.middleware.CSPMiddleware",) + MIDDLEWARE
+CSP_REPORT_URI = "https://your-sentry-url-prefix.com/api/{csp-project-id}/security/?sentry_key={sentry-key}"
+CSP_REPORT_ONLY = True
+```
```
Comment (mdtro) [inline src/docs/self-hosted/csp.mdx:16]:
```suggestion
```python
if "csp.middleware.CSPMiddleware" not in MIDDLEWARE:
    MIDDLEWARE = ("csp.middleware.CSPMiddleware",) + MIDDLEWARE
CSP_REPORT_URI = "https://your-sentry-url-prefix.com/api/{csp-project-id}/security/?sentry_key={sentry-key}"
CSP_REPORT_ONLY = True
```
```

File: src/docs/self-hosted/csp.mdx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+We recommend starting with `CSP_REPORT_ONLY = True`. Once there are no violations under normal use, you can switch it to `CSP_REPORT_ONLY = False`. This will enforce CSP and all unwanted scripts/resources will be blocked.
+
+In case you need to allow custom sources, you can add extra CSP keywords. We support [django-csp](https://django-csp.readthedocs.io/en/latest/configuration.html) syntax. For example:
```
Comment (hubertdeng123) [inline src/docs/self-hosted/csp.mdx:20]:
```suggestion
If you'd like to allow custom sources, extra CSP keywords can be added. We support the [django-csp](https://django-csp.readthedocs.io/en/latest/configuration.html) syntax. For example:
```

File: src/docs/self-hosted/csp.mdx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+---
+title: "Self-Hosted Content Security Policy (CSP)"
+---
+
+<Alert level="warning">
+  This is an experimental feature. Use it with caution.
+</Alert>
+
+Starting with Sentry `23.5.0`, it is possible to enable [CSP header](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) on the self-hosted Sentry installation. The good news is that Sentry itself supports [CSP reports collection](https://docs.sentry.io/product/security-policy-reporting/). We recommend creating a separate Sentry project for CSP reports. To enable CSP and reports collection, you'll want to configure the following settings in `sentry.conf.py`:
```
[the remark this answers is not in the document]
Comment (hubertdeng123) [reply src/docs/self-hosted/csp.mdx:9]:
```suggestion
Starting with Sentry `23.5.0`, it is possible to enable [CSP header](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) on self-hosted Sentry installations. The good news is that Sentry itself supports [CSP reports collection](https://docs.sentry.io/product/security-policy-reporting/). We recommend creating a separate Sentry project for CSP reports. To enable CSP and reports collection, you'll want to configure the following settings in `sentry.conf.py`:
```

**Comment by oioki** [comment]
I'd like to merge it next week after 23.5.0 release.

**Pull request merged**